File: CS/CM/CMWeb/App_Code/CMCommonService.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Activation;
using System.ServiceModel.Web;
using System.Text;

using Seasar.Quill;

using NEXS.ERP.CM.Common;
using NEXS.ERP.CM.BL;

//************************************************************************
/// <summary>
/// 共通検索AJAXサービス
/// </summary>
//************************************************************************
[ServiceContract(Namespace = "")]
[AspNetCompatibilityRequirements(RequirementsMode = AspNetCompatibilityRequirementsMode.Allowed)]
public class CMCommonService
{
    #region インジェクション用フィールド
    protected ICMCommonBL m_commonBL;
    #endregion

    #region コンストラクタ
    //************************************************************************
    /// <summary>
    /// コンストラクタ
    /// </summary>
    //************************************************************************
    public CMCommonService()
    {
        // インジェクション実行
        QuillInjector injector = QuillInjector.GetInstance();
        injector.Inject(this);
    }
    #endregion

    #region サービスメソッド
    //************************************************************************
    /// <summary>
    /// コード値から名称を取得する。
    /// </summary>
    /// <param name="argCode">コード値</param>
    /// <param name="argSelectId">共通検索ID</param>
    /// <param name="argSelectParam">共通検索パラメータ</param>
    /// <returns>コード値に対する名称</returns>
    //************************************************************************
    [OperationContract]
    [WebGet]
    public CodeName GetCodeName(string argCode, string argSelectId, string argSelectParam)
	{
        string name = "";

        List<object> paramList = new List<object>();
        paramList.Add(argCode);

        // 共通検索パラメータ作成
        if (!string.IsNullOrEmpty(argSelectParam))
        {
            foreach (string p0 in argSelectParam.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries))
            {
                string p = p0.TrimStart();
                if (p.Length < 2) continue;

                // 'から始まる場合はそのまま設定
                if (p[0] == '\'') paramList.Add(p.Substring(1));
                // "#"から始まる場合はUserInfoから設定
                else if (p[0] == '#')
                {
                    System.Reflection.PropertyInfo pi = CMInformationManager.UserInfo.GetType().GetProperty(p.Substring(1));
                    paramList.Add(pi.GetValue(CMInformationManager.UserInfo, null));
                }
                // Rowの値を取得
                //else paramList.Add(row[p]);
            }
        }

        // 検索実行
        DataTable result = m_commonBL.Select(argSelectId, paramList.ToArray());
        if (result != null && result.Rows.Count > 0)
            name = result.Rows[0][0].ToString();

        // 結果を返却
        return new CodeName() { Name = name };
    }
    #endregion

    //************************************************************************
    /// <summary>
    /// コード値に対する名称返却クラス
    /// </summary>
    //************************************************************************
    public class CodeName
    {
        /// <summary>コード値に対する名称</summary>
        public string Name { get; set; }
    }
}

File: VB/CM/CMWeb/App_Code/CMCommonService.vb
Imports System.Data
Imports System.ServiceModel
Imports System.ServiceModel.Activation
Imports System.ServiceModel.Web

Imports Seasar.Quill

Imports NEXS.ERP.CM.Common
Imports NEXS.ERP.CM.BL

''' <summary>
''' 共通検索AJAXサービス
''' </summary>
<ServiceContract(Namespace:="")>
<AspNetCompatibilityRequirements(RequirementsMode:=AspNetCompatibilityRequirementsMode.Allowed)>
Public Class CMCommonService
#Region "インジェクション用フィールド"
    Protected m_commonBL As ICMCommonBL
#End Region

#Region "コンストラクタ"
    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
        ' インジェクション実行
        Dim injector As QuillInjector = QuillInjector.GetInstance()
        injector.Inject(Me)
    End Sub
#End Region

#Region "サービスメソッド"
    ''' <summary>
    ''' コード値から名称を取得する。
    ''' </summary>
    ''' <param name="argCodeId">コード値要素ID属性</param>
    ''' <param name="argNameId">名称要素ID属性</param>
    ''' <param name="argSelectId">共通検索ID</param>
    ''' <param name="argCode">コード値</param>
    ''' <returns>コード値に対する名称</returns>
    <OperationContract()> _
    <WebGet()> _
    Public Function GetCodeName(argCodeId As String, argNameId As String, argSelectId As String, argCode As String) As CodeName
        Dim name As String = ""

        ' 検索実行
        Dim result As DataTable = m_commonBL.[Select](argSelectId, argCode)
        If result IsNot Nothing AndAlso result.Rows.Count > 0 Then
            name = result.Rows(0)(0).ToString()
        End If

        ' 結果を返却
        Return New CodeName() With {.CodeId = argCodeId, .NameId = argNameId, .Name = name}
    End Function
#End Region

    ''' <summary>
    ''' コード値に対する名称返却クラス
    ''' </summary>
    Public Class CodeName
        ''' <summary>コード値要素ID属性</summary>
        Public Property CodeId() As String

        ''' <summary>名称要素ID属性</summary>
        Public Property NameId() As String

        ''' <summary>コード値に対する名称</summary>
        Public Property Name() As String
    End Class
End Class

